src/modules/system/System.h

```diff
@@ -102,9 +102,10 @@ class System : public Module
 	/**
 	 * Vibrates for the specified amount of seconds.
 	 *
-	 * @param number of seconds to vibrate.
+	 * @param seconds number of seconds to vibrate.
+	 * @param mode vibration mode.
 	 */
-	virtual void vibrate(double seconds) const;
+	virtual void vibrate(double seconds, const std::string &mode) const;
```

Comment:
I think this should use an enum here (like the PowerState enum in this file, if you need an example to work from) instead of passing the string all the way down to the platform implementation.